src/core/net/srp_server.cpp

```diff
@@ -438,15 +438,32 @@ void Server::CommitSrpUpdate(Error                    aError,
 
 void Server::Start(void)
 {
-    Error error = kErrorNone;
+    Error    error = kErrorNone;
+    uint16_t port  = kUdpPortMin;
 
-    VerifyOrExit(!IsRunning());
+#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE && OPENTHREAD_CONFIG_SRP_SERVER_SAVE_INFO
+    Settings::SrpServerInfo info;
+    if (Get<Settings>().Read(info) == kErrorNone)
+    {
+        port = info.GetPort() + 1;
+        if (port > kUdpPortMax)
+        {
+            port = kUdpPortMin;
+        }
+    }
+#endif
 
+    VerifyOrExit(!IsRunning());
```

Comment:
How about doing `VerifyOfExit(!IsRunning())` at the top of the method and before reading the info from `Settings`?